[DOCS] Rules page: advanced sorting and bulk editing (#1621)
elastic/security-docs: docs/detections/rules-ui-manage.asciidoc

[role="xpack"]
== Manage detection rules

The Rules page allows you to view and manage all prebuilt and custom detection rules.

[role="screenshot"]
image::images/all-rules.png[The Rules page]

> You can sort the rules by clicking on the *Rule*, *Last updated*, or *Enabled* column header. 

Reviewer: ```suggestion
You can sort the rules by clicking the *Rule*, *Last updated*, or *Enabled* column header. 
```

> TIP: To sort by any other column, enable the *Technical preview* toggle above the table. This experimental table view allows advanced sorting capabilities. If you experience performance issues when working with the table, you can turn this setting off.

Reviewer: A UX writing thing - we are trying to steer away from "enable/disable the toggle". 

```suggestion
TIP: To sort by any other column, switch the *Technical preview* toggle on above the table. This experimental table view allows advanced sorting capabilities. If you experience performance issues when working with the table, you can turn this setting off.
```

image::images/install-prebuilt-rules.png[]

[float]
[[edit-rules-settings]]
=== Modify existing rules settings

> You can edit the settings of an existing rule. You can also select multiple rules, then edit certain settings in bulk (index patterns and tags).

Reviewer 2: ```suggestion
You can edit an existing rule's settings, and can edit index patterns and tags for multiple rules at once.
```

NOTE: For prebuilt Elastic rules, you can't modify most settings. You can only edit <<rule-schedule, rule actions>> and <<detections-ui-exceptions, add exceptions>>.

. Go to *Detect* -> *Rules*.
. Do one of the following:

> * Edit a single rule: Click the *All actions* icon (*...*) on a rule, then select *Edit rule settings*. The *Edit rule settings* view opens, where you can modify the <<rules-ui-create, rule's settings>>.

Reviewer: This is a UX thing; this is technically considered a menu. 
```suggestion
* Edit a single rule: Select the *All actions* menu (*...*) on a rule, then select *Edit rule settings*. The *Edit rule settings* view opens, where you can modify the <<rules-ui-create, rule's settings>>.
```

* Edit multiple rules: Select all the rules you want to edit, then select an action from the *Bulk actions* menu:
** *Index patterns*: Add or delete the index patterns used by all selected rules.
** *Tags*: Add or delete tags on all selected rules.

> . On the flyout that opens, enter the settings to change on the rules.

Reviewer 2: ```suggestion
. On the flyout that opens, update the settings.
```

. If available, select *Overwrite all selected _x_* to overwrite the settings on the rules. For example, if you're adding tags to multiple rules, selecting *Overwrite all selected rules tags* removes all the rules' original tags and replaces them with the tags you specify.
. Click *Save*.

[float]
[[manage-rules-ui]]
=== Manage rules

You can duplicate, enable, disable, and delete rules:

. Go to *Detect* -> *Rules*.
. Do one of the following:

> * Click the *All actions* icon (*...*) on a rule, then select an action.

Reviewer: ```suggestion
* Select the *All actions* menu (*...*) on a rule, then select an action.
```

> * Click the *All actions* icon (*...*) on a rule, then select the required action.

Reviewer 2: ```suggestion
* Click the *All actions* icon (*...*) on a rule, then select the desired action.
```

Reviewer 2: just putting it out there — to me required sounds like we are requiring them to do it. Probably not confusing in this context but I still like desire 🤷

Author: @benironside I'm not a fan of "desire" in technical docs, but definitely agree that "required" isn't right. I think we can omit an adjective altogether :
```suggestion
* Click the *All actions* icon (*...*) on a rule, then select an action.
```

* Select all the rules you want to modify, then select an action from the *Bulk actions* menu.

> * To enable or disable a single rule, click the rule's *Enabled* toggle.

Reviewer: ```suggestion
* To enable or disable a single rule, switch the rule's *Enabled* toggle on.
```

> * To enable or disable a single rule, click the *Enabled* toggle on the rule.

Reviewer 2: ```suggestion
* To enable or disable a single rule, click the rule's *Enabled*/*Disabled* toggle.
```

Author: The toggle label actually never changes to *Disabled* so it's just *Enabled*. But I like the other edit for brevity:
```suggestion
* To enable or disable a single rule, click the rule's *Enabled* toggle.
```
